### PR TITLE
[FEATURE] Stable types for `@ember/polyfill`

### PR DESCRIPTION
**Pull request by theroncross**
toward #19946

### CODE REVIEW

**Review by chriskrycho — CHANGES_REQUESTED** [review]
Excellent! This also needs the corresponding entries (and files!) removed from the preview types. Once you do that, ping me and I'll re-review and land it. Thank you!

**Comment by theroncross** [comment]
Ah, that step is missing from https://github.com/emberjs/ember.js/issues/19946#issuecomment-1330635787

In this case, the preview types are an improvement over the source types. However, they have a dependency on the `Mix` types from `/types/preview/@ember/polyfills/types.d.ts`, which don't seem to be used anywhere else. Do you have a preference for where one-off utility types live?

**Comment by chriskrycho** [comment]
@theroncross are you still up to finish updating this? Would love to land it!

**Comment by theroncross** [comment]
Sorry about that. Got sidetracked by life a bit.

The tests needed some changes to match the `Object.assign` type behavior, but it should be close now.

**Comment by chriskrycho** [comment]
Sweet, will take a look tomorrow, and just re-approved CI! Thank you!

File: packages/@ember/polyfills/lib/assign.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { deprecate } from '@ember/debug';
 export function assign<T, U>(target: T, source: U): T & U;
 export function assign<T, U, V>(target: T, source1: U, source2: V): T & U & V;
 export function assign<T, U, V, W>(target: T, source1: U, source2: V, source3: W): T & U & V & W;
-export function assign(target: object, ...sources: object[]): object;
+export function assign(target: object, ...sources: any[]): any;
```
Comment (chriskrycho) [inline packages/@ember/polyfills/lib/assign.ts:10]: :sigh: This is probably the “right” thing to do here, since it’s what TS itself has, but I hate it.

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
@theroncross looks good overall but I think some of the type tests will need updating!

**Comment by chriskrycho** [comment]
@theroncross I think we're going to end up closing this as "not needed" since it's all removed as of Ember v5. Thank you again, though!

**Comment by chriskrycho** [comment]
Superseded here specifically by #20380.

**Pull request closed without merging**